### PR TITLE
tests: drivers: build_all: adc: Add spi-devices build test

### PR DESCRIPTION
**Pull request by soburi**
Add build tests for following devices.

- ti,tla2021

- - -
And ADS1112 was not enabled correctly, so I fixed it.

### CODE REVIEW

File: tests/drivers/build_all/adc/boards/native_sim.overlay
Comment (decsny) [inline tests/drivers/build_all/adc/boards/native_sim.overlay:1]: please make commit title more specific about this device

Comment (decsny) [inline tests/drivers/build_all/adc/boards/native_sim.overlay:1]: please make commit title more specific about this device

Comment (soburi) [reply tests/drivers/build_all/adc/boards/native_sim.overlay:1]: Update done.
Thank you for your check.

**Review by decsny — DISMISSED** [review]

**Comment by soburi** [comment]
@anangl 
Could you take a look if you have time?

File: tests/drivers/build_all/adc/boards/native_sim.overlay [hunk 1]
```diff
@@ -344,6 +344,13 @@
 					#io-channel-cells = <1>;
 				};
 			};
+
+			test_spi_tla2021: tla2021@20 {
+				compatible = "ti,tla2021";
```
Comment (anangl) [inline tests/drivers/build_all/adc/boards/native_sim.overlay:349]: This is an I2C device, not an SPI one.

**Review by anangl — CHANGES_REQUESTED** [review]

**Review by anangl — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Pull request merged**